### ISSUE TITLE
[Fix] installing Rhdf5lib on M1/M2 OSX with conda version of R 

### ISSUE THREAD

**Issue by eyalbenda** [grimbough/Rhdf5lib#54]
So, after a long and miserable time trying to get Rhdf5lib installed on the M1 OSX (binary version couldn't work for me) I traced the issue and found a solution. Note, this has been tested by me using the conda-forge version of R(4.3.1).

See: #45 #51 

apparently hdf5's configure script gets the build alias from $host_alias or $build_alias if they're defined. The following command allowed a clean install from inside R with BiocManager::install("Rhdf5lib")

> unset host_alias 
> unset build_alias

I'm putting it here as a fix. This could potentially be fixed by the package's install script, for your consideration.

**Comment by grimbough** [comment]
Thanks for the report @eyalbenda 

Just so I'm clear, what was the symptom of the issue for you?  The error messages in #45 and #51 look different to me.

**Comment by eyalbenda** [comment]
That’s because #51 already tries to install the binary, which isn’t
installed correctly on conda R builds. Unsetting the flags allows
installation from source, solving #45 and obviating the need for binary
installation.

**Comment by wakelin-g** [comment]
I can confirm that with conda-forge R (4.3.2), this is still an issue and this fix continues to work.

```
R version 4.3.2 (2023-10-31)
Platform: aarch64-apple-darwin20.0.0 (64-bit)
Running under: macOS Sonoma 14.2.1

Matrix products: default
BLAS/LAPACK: /Users/griffen/mambaforge/envs/r-osca/lib/libopenblas.0.dylib;  LAPACK version 3.12.0

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

time zone: America/Halifax
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] Rhdf5lib_1.24.1

loaded via a namespace (and not attached):
[1] BiocManager_1.30.22 compiler_4.3.2      tools_4.3.2
```

**Comment by grimbough** [comment]
Now GitHub have made and M1 Mac available as an Actions runner it much more likely I'll be able to reproduce and test this.  I'll report back here when I have done so.

**Comment by grimbough** [comment]
I think this should now be sorted.  I updated some of the scripts bundled with the HDF5 source which were quite out of date and didn't recognise the combination of conda and arm64 Mac.  It's available in Rhdf5lib version 1.24.2, which I guess might take a few days to be available via the Bioconductor repository.
